src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -0,0 +1,239 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "opt.hpp"
+
+#include "../../logging.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reduce_sum.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/split.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/pass/pattern/op/label.hpp"  // any_input
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/util/common_util.hpp"
+
+// FIXME: It is probably much better to have these transformations done
+// just once per function BODY
+// On LLMs, it may reduce the # of such transformations by a factor of
+// 20..30x.
+
+namespace ov {
+namespace npuw {
+namespace patterns {
+namespace opt {
```

Comment:
nit: since C++17 this can be written `namespace ov::npuw::patterns::opt {`

Reply:
cool though I thought the plugin is still c++14?